tokio/src/sync/once_cell.rs

```diff
@@ -182,11 +191,39 @@ impl<T> OnceCell<T> {
     // and tokio MSRV is bumped to the rustc version with it stabilised,
     // we can make this function available in const context,
     // by creating `Semaphore::const_new_closed`.
+    #[track_caller]
     pub fn new_with(value: Option<T>) -> Self {
-        if let Some(v) = value {
-            OnceCell::from(v)
-        } else {
-            OnceCell::new()
+        #[cfg(all(tokio_unstable, feature = "tracing"))]
+        let resource_span = {
+            let location = std::panic::Location::caller();
+
+            tracing::trace_span!(
+                parent: None,
+                "runtime.resource",
+                concrete_type = "OnceCell",
+                kind = "Sync",
+                loc.file = location.file(),
+                loc.line = location.line(),
+                loc.col = location.column(),
+            )
+        };
+        #[cfg(all(tokio_unstable, feature = "tracing"))]
+        resource_span.in_scope(|| {
+            tracing::trace!(
+                target: "runtime::resource::state_update",
+                state = "unset",
+            );
+        });
```

Comment:
Although it's nice to add tracing support, this PR is already complex enough that I would prefer to add it in a follow-up PR.

Reply:
I've removed it in https://github.com/tokio-rs/tokio/pull/7161/commits/7409540229dede17601e4480b60ef5af7e5dc84c